microsoft/PowerToys
# [FancyZones] Move window into last known position on active work area (if possible)

## Summary of the Pull Request
Since we are supporting app zone history on multiple work areas (virtual desktops, monitors) we need to distinguish on which monitor window should be opened (within same virtual desktop) and try to zone it there based on app zone history, if possible:
1. Use monitor on which cursors is currently possitioned and virtual desktop on which is window that we want to zone. If in that work area we have zone history for this process, zone it.
2. If we failed to zone window in previous step for any reason, fallback to previosus solution, which is primary monitor (windows is always opened on primary monitor).


## PR Checklist
* [x] Applies to #4182


## Validation Steps Performed
1. Two (or more) monitor setup. Create two (or more) virtual desktops.
2. Open several instances of same application and zone it on different monitors / virtual desktops.
3. Check `C:\Users\<user-name>\AppData\Local\Microsoft\PowerToys\FancyZones\app-zone-history.josn` if zone history is valid.
4. Close those instances, and reopen them again on different monitors / virtual desktops.

Expected result: Window is placed into last known zone. Monitor itself is being determined based on cursor position (similar to opening FZ editor).
